### PR TITLE
feat(builder): Block UI Types and StickyNote

### PR DESCRIPTION
**Pull request by Swiftyos**
### Background

We need to be able to render Custom Node UI's for specific blocks, such as Input, Output and StickyNotes. 

### Changes 🏗️

This PR adds:
- an annotation to blocks that specifies the UI to render, with a default of standard. 
- a sticky note block, with a text input, no output and no run logic.

___

### **PR Type**
enhancement

___

### **Description**
- Introduced a new `BlockUIType` enum to specify UI types for blocks, including standard, input, output, and sticky note.
- Added `ui_type` attribute to the `Block` class to define the UI type for each block.
- Implemented a new `StickyNoteBlock` class with a text input, no output, and a specific UI type.
- Updated `InputBlock` and `OutputBlock` classes to include UI type specification.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>basic.py</strong><dd><code>Add UI type specification and StickyNoteBlock class</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/autogpt_server/blocks/basic.py

<li>Added <code>BlockUIType</code> to <code>InputBlock</code> and <code>OutputBlock</code> classes.<br> <li> Introduced <code>StickyNoteBlock</code> class with a text input and no output.<br> <li> Set UI type for blocks using <code>BlockUIType</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7994/files#diff-fd165de859043484585a7af9f38dcca0fd532ce08acfb3c8984fe0f08596e485">+36/-3</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>block.py</strong><dd><code>Introduce BlockUIType enum and update Block class</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/autogpt_server/data/block.py

<li>Introduced <code>BlockUIType</code> enum for block UI types.<br> <li> Added <code>ui_type</code> attribute to <code>Block</code> class.<br> <li> Updated <code>to_dict</code> method to include <code>uiType</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7994/files#diff-146658c7d8e69e8d71aa7eb64f0950731ad5a48ca3eb7d5d2e5225dabebe08f1">+10/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: rnd/autogpt_server/autogpt_server/data/block.py [hunk 1]
```diff
@@ -16,6 +16,13 @@
 CompletedBlockOutput = dict[str, list[Any]]  # Completed stream, collected as a dict.
 
 
+class BlockUIType(Enum):
+    STANDARD = "Standard"
```
Comment (ntindle) [inline rnd/autogpt_server/autogpt_server/data/block.py]: micro nit: Can we come up with some more clear name here than standard? It works but :(

Comment (Swiftyos) [reply rnd/autogpt_server/autogpt_server/data/block.py]: I had also considered Default and Normal. What do you think?

Comment (ntindle) [reply rnd/autogpt_server/autogpt_server/data/block.py]: Is there a better name for the collection that describes them? Inputs is what I’d choose but that has other meanings

Comment (Swiftyos) [reply rnd/autogpt_server/autogpt_server/data/block.py]: I don't understand what you are getting at.

Comment (aarushik93) [reply rnd/autogpt_server/autogpt_server/data/block.py:24]: what would the alternatives to Default be as an example?

Comment (aarushik93) [reply rnd/autogpt_server/autogpt_server/data/block.py]: what would default look like in the UI?

Comment (Swiftyos) [reply rnd/autogpt_server/autogpt_server/data/block.py]: How it currently is a normal node

**Pull request merged**